### DISCUSSION TITLE
How to use Boolean trigger

### DISCUSSION THREAD

**Discussion by sfabry** [alexzhornyak/SCXML-tutorial#18] [Q&A] [answered]
I have following simple test:
![image](https://user-images.githubusercontent.com/3390034/158144389-21bf60ad-24d3-46d9-8bdf-787f8e2cf222.png)

Using an integer trigger is working. 
The self connection on top is only intended to add logs, but it will not trigger when the conditionnal triggers. I have only logs when we are and stay in CenterLaserOn state. Is it normal ?

I can trigger this transition from Qml with following code: 
`        onOnChanged: root.stateMachine.submitEvent("CenterLaserToggled", on ? 1 : 0)`

Now what if wanna use a boolean trigger:
![image](https://user-images.githubusercontent.com/3390034/158144980-db474229-ceba-47b0-85cd-fffabdcfa4f2.png)
It seems to make the trigger work that the 'true/false' is in TEXT which is not the best idea.
How to use this boolean trigger to make state switch from OFF to ON.

Further I wanna use this code from Qml:
'        onOnChanged: root.stateMachine.submitEvent("ExitInLaserToggled", on)'

But it does not pass the condition, obviously because the test is on 'true' TEXT, how to use it on boolean value ?

Thanks

**Comment by sfabry** [comment, marked as answer]
Bonus question: when i use the statemachine from QML and link to the scxmlEditor, I can see state changes in the editor but i don't have the logs anymore... Any idea ?

**Reply by alexzhornyak** [reply]
Unfortunately there is a well-known bug in Qt [https://bugreports.qt.io/browse/QTBUG-92108](https://bugreports.qt.io/browse/QTBUG-92108)
So, Qt always sends a string in EcmaScript with submitEvent, so I offer for you not to use bool, send integer 1 or 0 and compare with '=='

Here I will show how it works with UScxml and in Qt

![issue_bool](https://user-images.githubusercontent.com/18611095/158149007-87e9af5a-b0c1-46d6-9960-751aa4ccf800.gif)

**Reply by alexzhornyak** [reply]
> Bonus question: when i use the statemachine from QML and link to the scxmlEditor, I can see state changes in the editor but i don't have the logs anymore... Any idea ?

You will see it in Qt Creator

![image](https://user-images.githubusercontent.com/18611095/158150437-21885232-22e0-48cc-82d5-77bbc222c838.png)

**Comment by sfabry** [comment]
Thank you very much !!

And congrats for the Editor software, it's amazing :)

**Comment by alexzhornyak** [comment]
> The self connection on top is only intended to add logs, but it will not trigger when the conditionnal triggers. I have only logs when we are and stay in CenterLaserOn state. Is it normal ?

Your top connection is incorrect because it is intercepted by inner.
